treeherder/client/thclient/client.py

```diff
@@ -684,6 +686,14 @@ def _get_uri(self, project, endpoint, data=None, oauth_key=None,
 
         return uri
 
+    def _get_revision_id(self, repo_name, revision):
```

Comment:
Let's make this a public method, it's more generally useful. Also, repo_name should be project.